challenge-manager/challenges.go

```diff
@@ -83,61 +120,62 @@ func (m *Manager) ChallengeAssertion(ctx context.Context, id protocol.AssertionH
 	log.Info("Successfully opened a challenge on an invalid assertion",
 		"name", m.name,
 		"assertionHash", containers.Trunc(id.Bytes()),
-		"fromBatch", edgeTrackerAssertionInfo.FromBatch,
-		"toBatch", edgeTrackerAssertionInfo.ToBatch,
+		"fromBatch", edgeTrackerAssertionInfo.FromState.Batch,
+		"fromPosInBatch", edgeTrackerAssertionInfo.FromState.PosInBatch,
+		"batchLimit", edgeTrackerAssertionInfo.BatchLimit,
 	)
 	return true, nil
 }
 
 func (m *Manager) addBlockChallengeLevelZeroEdge(
 	ctx context.Context,
 	assertion protocol.Assertion,
-) (protocol.VerifiedRoyalEdge, bool, *edgetracker.AssociatedAssertionMetadata, bool, error) {
+) (protocol.VerifiedRoyalEdge, bool, *l2stateprovider.AssociatedAssertionMetadata, bool, error) {
 	creationInfo, err := m.chain.ReadAssertionCreationInfo(ctx, assertion.Id())
 	if err != nil {
 		return nil, false, nil, false, errors.Wrap(err, "could not get assertion creation info")
 	}
-	if !m.allowTrackingEdgeWithChallengeParentAssertionHash(protocol.AssertionHash{Hash: creationInfo.ParentAssertionHash}) {
+	if !m.watcher.AllowTrackingEdgeWithParentHash(protocol.AssertionHash{Hash: creationInfo.ParentAssertionHash}) {
```

Comment:
Side note: the ergonomics of these custom types, such as assertion hash, haven't been that nice to use. It is cumbersome to write `protocol.AssertionHash{Hash: creationInfo.ParentAssertionHash}` a lot in the codebase and perhaps we want to improve its usage in the future

Reply:
My thinking here is that we probably aren't pushing the type conversions far enough to the edges of our data processing.  For example, in this specific case, why not have the `m.chain.ReadAssertionCreationInfo` return a type that has already wrapped the `common.Hash` values in `protocol.AssertionHash` instances?

Reply:
Checkout [0fb69b9](https://github.com/OffchainLabs/bold/pull/697/commits/0fb69b93b9b6482434afd705dc6e8e0d967c2f82) which fixed this up very nicely IMHO.